File: common/src/main/java/juuxel/adorn/util/animation/AnimatedProperty.java
package juuxel.adorn.util.animation;

public final class AnimatedProperty<T> extends AbstractAnimatedProperty<T> {
    private T value;

    public AnimatedProperty(T initial, AnimationEngine engine, int duration, Interpolator<T> interpolator) {
        super(engine, duration, interpolator);
        value = initial;
    }

    @Override
    public T get() {
        return value;
    }

    @Override
    protected void setRawValue(T value) {
        this.value = value;
    }
}

File: common/src/main/kotlin/juuxel/adorn/util/animation/AnimatedProperty.kt
package juuxel.adorn.util.animation

class AnimatedProperty<T>(
    initial: T,
    engine: AnimationEngine,
    duration: Int,
    interpolator: Interpolator<T>
) : AbstractAnimatedProperty<T>(engine, duration, interpolator) {
    private var value: T = initial

    override fun get(): T = value
    override fun setRawValue(value: T) {
        this.value = value
    }
}
